pkg/executor/delete.go

```diff
@@ -164,15 +167,31 @@ func (e *DeleteExec) doBatchDelete(ctx context.Context) error {
 	return nil
 }
 
+// fixHandlePosInfoForMultiDelete fixes the position info by the current execution logic.
+// We will cut the mixed row into single table row, so the handle position info should be fixed.
+//   e.g. The multi-delete case is [t1.a, t1.b, t2.a, t2.b] and There's a index [t2.b].
+//        The idxCols is [3]. And we should fix it to [1].
+func (e *DeleteExec) fixHandlePosInfoForMultiDelete() {
```

Comment:
Why don't we do this in the planner? We already have enough information and are doing similar things there.